src/__tests__/to-have-value.js

```diff
@@ -82,6 +82,41 @@ describe('.toHaveValue', () => {
     expect(queryByTestId('single')).toHaveValue('first')
   })
 
+  test('handles values of custom select with role `listbox`', () => {
+   const {queryByTestId} = render(`
+     <div role="listbox" data-testid="single">
+       <div role="option" value="first">First Value</div>
+       <div role="option" value="second" aria-selected="true">Second Value</div>
+       <div role="option" value="third">Third Value</d>
+     </div>
+ 
+     <div role="listbox" data-testid="multiple" aria-multiselectable>
+       <div role="option" value="first">First Value</div>
+       <div role="option" value="second" aria-selected="true">Second Value</div>
+       <div role="option" value="third" aria-selected="true">Third Value</div>
```

Comment:
I'm not sure that `value` is a standard attribute of `div` elements. I'm not even sure this pattern of `role="option" value="…"` is documented or valid.

Reply:
Hi @gnapse, thank's by feedback.

https://www.w3.org/TR/wai-aria-1.1/#option, this is ARIA documentation oficial, and specified base concept of role "option", is a option element; https://html.spec.whatwg.org/multipage/form-elements.html#the-option-element, I'm aware that the documentation does not explicitly say that you need an attribute named `value`, maybe a good option to use would be` data-value`. 

`<div role="option" data-value="first">First value</div>`

This a example why is necesary, Here is a sandbox, which illustrates the problem, with an input type = "hidden", the access would not be so transparent, and it would be necessary to obtain this to make the assertion of the value. 

https://codesandbox.io/s/bold-lehmann-cwfw0?file=/tests/CustomSelect.test.js

Reply:
I understand your concern and your need. I've worked both implementing custom form-like components, and used them.

However, my concern here is that this is a generic DOM assertions library. The approach you use to define the "value" of each of this custom option elements is specific to your app. One library may be using this `value` attribute (which is strictly invalid) while another library may be using the more acceptable `data-value` to keep this data in the DOM. But this assertion library should not pick sides and implement a custom assertion that assumes one of this app-specific solutions to this problem.

That's why I suggested that this may not be needed at all. What I really think a component like this should do is to provide a `<input type="hidden" value="…" /> that's kept up-to-date with what the current value is.

Reply:
I understand, if is necessary to modify the pull request, so that suits the needs of the library, feel free to let me know, and if you need to close the pull request it is also fine, in my case, I think I managed to make the assertion directly in the `<div role =" option> `, which in` <div role = "listbox"> `, I just thought it necessary to contribute, thank's so much for the review. Regards